k9mail/src/main/java/com/fsck/k9/extensions/LocalMessage.kt

```diff
@@ -13,6 +15,16 @@ fun LocalMessage.isMessageIncomplete(): Boolean {
     return !isSet(Flag.X_DOWNLOADED_FULL) && !isSet(Flag.X_DOWNLOADED_PARTIAL)
 }
 
+fun LocalMessage?.isValidForHandshake() = this != null
+        && (account?.isPlanckPrivacyProtected ?: false)
+        && from != null
+        && from.size == 1
+        && PlanckUtils.isRatingReliable(planckRating)
+        && getRecipients(Message.RecipientType.CC).isNullOrEmpty()
+        && getRecipients(Message.RecipientType.BCC).isNullOrEmpty()
+        && !getRecipients(Message.RecipientType.TO).isNullOrEmpty()
+        && !from.first().address.equals(account.email, ignoreCase = true) // sender not my own account
```

Comment:
could be extracted to the dedicated function

Reply:
which part? since this is already a function

Reply:
Line 26

Reply:
I would not extract that line alone, it is missing context of nulllability etc.